bridges/OpenlyBridge.php

```diff
@@ -0,0 +1,225 @@
+<?php
+class OpenlyBridge extends BridgeAbstract {
+	const NAME = 'Openly Bridge';
+	const URI = 'https://www.openlynews.com/';
+	const DESCRIPTION = 'Returns news articles';
+	const MAINTAINER = 'VerifiedJoseph';
+	const PARAMETERS = array(
+		'All News' => array(),
+		'All Opinion' => array(),
+		'By Region' => array(
+			'region' => array(
+				'name' => 'Region',
+				'type' => 'list',
+				'values' => array(
+					'Africa' => 'africa',
+					'Asia Pacific' => 'asia-pacific',
+					'Europe' => 'europe',
+					'Latin America' => 'latin-america',
+					'Middle Easta' => 'middle-east',
+					'North America' => 'north-america'
+				)
+			),
+			'content' => array(
+				'name' => 'Content',
+				'type' => 'list',
+				'values' => array(
+					'News' => 'news',
+					'Opinion' => 'people'
+				),
+				'defaultValue' => 'news'
+			)
+		),
+		'By Tag' => array(
+			'tag' => array(
+				'name' => 'Tag',
+				'type' => 'text',
+				'required' => true,
+				'exampleValue' => 'lgbt-law',
+			),
+			'content' => array(
+				'name' => 'Content',
+				'type' => 'list',
+				'values' => array(
+					'News' => 'news',
+					'Opinion' => 'people'
+				),
+				'defaultValue' => 'news'
+			)
+		),
+		'By Author' => array(
+			'profileId' => array(
+				'name' => 'Profile ID',
+				'type' => 'text',
+				'required' => true,
+				'exampleValue' => '003D000002WZGYRIA5',
+			)
+		)
+	);
+
+	const CACHE_TIMEOUT = 900; // 15 mins
+	const ARTICLE_CACHE_TIMEOUT = 3600; // 1 hour
+
+	private $feedTitle = '';
+	private $itemLimit = 10;
+
+	private $profileRegexUrl = '/openlynews\.com\/profile\/\?id=([a-zA-Z0-9]+)/';
+	private $tagRegexUrl = '/openlynews\.com\/([a-z]+)\/\?(?:page=(?:[0-9]+)&)?theme=([\w-]+)/';
+
+	public function detectParameters($url) {
+		$params = array();
+
+		if(preg_match($this->profileRegexUrl, $url, $matches) > 0) {
+			$params['context'] = 'By Author';
+			$params['profileId'] = $matches[1];
+			return $params;
+		}
+
+		if(preg_match($this->tagRegexUrl, $url, $matches) > 0) {
+			$params['context'] = 'By Tag';
+			$params['content'] = $matches[1];
+			$params['tag'] = $matches[2];
+			return $params;
+		}
+
+		return null;
+	}
+
+	public function collectData() {
+		$url = $this->getAjaxURI();
+
+		if ($this->queriedContext === 'By Author') {
+			$url = $this->getURI();
+		}
+
+		$html = getSimpleHTMLDOM($url)
+			or returnServerError('Could not request: ' . $url);
```

Comment:
I some time ago discovered, that "or returnServerError" is useless. getSimpleHTMLDOM already deals with non-200 http responses.